File: java/app-encryption/src/test/java/com/godaddy/asherah/appencryption/SuffixedPartitionTest.java
package com.godaddy.asherah.appencryption;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertTrue;
import static org.junit.jupiter.api.Assertions.assertFalse;

class SuffixedPartitionTest {

    private final static String testPartitionId = "test_partition_id";
    private final static String testServiceId = "test_service_id";
    private final static String testProductId = "test_product_id";
    private final static String testRegionSuffix = "test_region_suffix";

    private Partition partition;

    @BeforeEach
    void setUp() {
        partition = new SuffixedPartition(testPartitionId, testServiceId, testProductId, testRegionSuffix);
    }

    @Test
    void testGetSystemKeyId() {
        String systemKeyIdString = "_SK_" + testServiceId + "_" + testProductId + "_" + testRegionSuffix;
        assertEquals(systemKeyIdString, partition.getSystemKeyId());
    }

    @Test
    void testGetIntermediateKeyId() {
        String intermediateKeyIdString =
          "_IK_" + testPartitionId + "_" + testServiceId + "_" + testProductId + "_" + testRegionSuffix;
        assertEquals(intermediateKeyIdString, partition.getIntermediateKeyId());
    }

    @Test
    void testToString() {
        String toStringString = partition.getClass().getSimpleName() + "[partitionId=" + testPartitionId +
          ", serviceId=" + testServiceId + ", productId=" + testProductId + ", regionSuffix=" + testRegionSuffix + "]";
        assertEquals(toStringString, partition.toString());
    }

    @Test
    void testIsValidIntermediateKeyId() {
      String intermediateKeyIdStringSuffixed =
        "_IK_" + testPartitionId + "_" + testServiceId + "_" + testProductId + "_" + testRegionSuffix;
      assertTrue(partition.isValidIntermediateKeyId(intermediateKeyIdStringSuffixed));

      String intermediateKeyIdString =
        "_IK_" + testPartitionId + "_" + testServiceId + "_" + testProductId;
      assertTrue(partition.isValidIntermediateKeyId(intermediateKeyIdString));

      String invalidId =
        "_IK_some_other_partition" + "_" + testServiceId + "_" + testProductId;
      assertFalse(partition.isValidIntermediateKeyId(invalidId));
    }
}

File: csharp/AppEncryption/AppEncryption.Tests/AppEncryption/SuffixedPartitionTest.cs
using Xunit;

namespace GoDaddy.Asherah.AppEncryption.Tests.AppEncryption
{
    public class SuffixedPartitionTest
    {
        private const string TestPartitionId = "test_partition_id";
        private const string TestServiceId = "test_service_id";
        private const string TestProductId = "test_product_id";
        private const string TestSuffixRegion = "test_suffix_region";

        private readonly SuffixedPartition partition;

        public SuffixedPartitionTest()
        {
            partition = new SuffixedPartition(TestPartitionId, TestServiceId, TestProductId, TestSuffixRegion);
        }

        [Fact]
        private void TestSuffixedPartitionCreation()
        {
            Assert.NotNull(partition);
        }

        [Fact]
        private void TestGetPartitionId()
        {
            string actualTestPartitionId = partition.PartitionId;
            Assert.Equal(TestPartitionId, actualTestPartitionId);
        }

        [Fact]
        private void TestGetServiceId()
        {
            string actualServiceId = partition.ServiceId;
            Assert.Equal(TestServiceId, actualServiceId);
        }

        [Fact]
        private void TestGetProductId()
        {
            string actualProductId = partition.ProductId;
            Assert.Equal(TestProductId, actualProductId);
        }

        [Fact]
        private void TestGetSystemKeyId()
        {
            const string systemKeyIdString = "_SK_" + TestServiceId + "_" + TestProductId + "_" + TestSuffixRegion;
            Assert.Equal(systemKeyIdString, partition.SystemKeyId);
        }

        [Fact]
        private void TestGetIntermediateKeyId()
        {
            const string intermediateKeyIdString = "_IK_" + TestPartitionId + "_" + TestServiceId + "_" + TestProductId + "_" + TestSuffixRegion;
            Assert.Equal(intermediateKeyIdString, partition.IntermediateKeyId);
        }

        [Fact]
        private void TestToString()
        {
            string expectedToStringString = partition.GetType().Name + "[partitionId=" + TestPartitionId +
                    ", serviceId=" + TestServiceId + ", productId=" + TestProductId + ", regionSuffix=" + TestSuffixRegion + "]";
            Assert.Equal(expectedToStringString, partition.ToString());
        }

        [Fact]
        private void TestIsValidIntermediaKeyId()
        {
            const string intermediateKeyIdStringSuffixed = "_IK_" + TestPartitionId + "_" + TestServiceId + "_" + TestProductId + "_" + TestSuffixRegion;
            Assert.True(partition.IsValidIntermediateKeyId(intermediateKeyIdStringSuffixed));

            const string intermediateKeyIdStringSuffixedOther = "_IK_" + TestPartitionId + "_" + TestServiceId + "_" + TestProductId + "_" + "other_suffix";
            Assert.True(partition.IsValidIntermediateKeyId(intermediateKeyIdStringSuffixedOther));

            const string intermediateKeyIdString = "_IK_" + TestPartitionId + "_" + TestServiceId + "_" + TestProductId;
            Assert.True(partition.IsValidIntermediateKeyId(intermediateKeyIdString));

            const string invalidId = "_IK_some_other_partition" + "_" + TestServiceId + "_" + TestProductId;
            Assert.False(partition.IsValidIntermediateKeyId(invalidId));
        }
    }
}
